test/distributed/_composable/fsdp/test_fully_shard_state_dict.py

```diff
@@ -158,5 +165,59 @@ def _test_state_dict_save_load(self, model: nn.Module):
                 )
 
 
+class TestFullyShardStateDictMultiThread(FSDPTestMultiThread):
+    @property
+    def world_size(self):
+        return 2
+
+    @unittest.skipIf(not TEST_CUDA, "no cuda")
+    def test_rank0_offload_full_state_dict(self):
+        # Construct a reference unsharded model on all ranks
+        model_args = ModelArgs(dropout_p=0.0)
+        torch.manual_seed(42)
+        ref_model = Transformer(model_args).cuda()
+        for param in ref_model.parameters():
+            torch.distributed.broadcast(param.detach(), src=0)
```

Comment:
do we have some logic in `fully_shard` to broadcast the initial tensors?

Reply:
We do not right now. This broadcast here is actually because we are using MTPG, so I think that there is no way to ensure the same starting reference model on all ranks (since seed is per process, not per thread).